core/src/test/java/io/questdb/griffin/SqlParserTest.java

```diff
@@ -4974,6 +4974,104 @@ public void testEmptyWhere() throws Exception {
+        );
+    }
+
+    @Test
+    public void testNoopGroupByFailureWhenMissingColumn() throws Exception {
+        assertFailure(
+                "select sym, avg(bid) avgBid from x where sym in ('AA', 'BB' ) group by ",
+                "create table x (\n" +
+                        "    sym symbol,\n" +
+                        "    bid int,\n" +
+                        "    ask int\n" +
+                        ")  partition by NONE",
+                71,
+                "literal expected"
+        );
+    }
+
+    @Test
+    public void testNoopGroupByAfterFrom() throws Exception {
+        assertQuery(
+                "select-group-by sym, avg(bid) avgBid from (select [sym, bid] from x timestamp (ts))",
+                "select sym, avg(bid) avgBid from x group by sym",
+                modelOf("x")
+                        .col("sym", ColumnType.SYMBOL)
+                        .col("bid", ColumnType.INT)
+                        .col("ask", ColumnType.INT)
+                        .timestamp("ts")
+        );
+    }
+
+    @Test
+    public void testNoopGroupByFailureWhenUsing1KeyInSelectStatementBut2InGroupBy() throws Exception {
+        assertFailure(
+                "select sym1, avg(bid) avgBid from x where sym1 in ('AA', 'BB' ) group by sym1, sym2",
+                "create table x (\n" +
+                        "    sym1 symbol,\n" +
+                        "    sym2 symbol,\n" +
+                        "    bid int,\n" +
+                        "    ask int\n" +
+                        ")  partition by NONE",
+                0,
+                "group by column does not match key column is select statement "
+        );
+    }
+
+    @Test
+    public void testNoopGroupByFailureWhenUsing2KeysInSelectStatementButOnlyOneInGroupBy() throws Exception {
+        assertFailure(
+                "select sym1, sym2, avg(bid) avgBid from x where sym1 in ('AA', 'BB' ) group by sym1",
+                "create table x (\n" +
+                        "    sym1 symbol,\n" +
+                        "    sym2 symbol,\n" +
+                        "    bid int,\n" +
+                        "    ask int\n" +
+                        ")  partition by NONE",
+                0,
+                "group by column does not match key column is select statement "
+        );
+    }
+
+    @Test
+    public void testNoopGroupByFailureWhenUsingInvalidColumn() throws Exception {
+        assertFailure(
+                "select sym, avg(bid) avgBid from x where sym in ('AA', 'BB' ) group by badColumn",
+                "create table x (\n" +
+                        "    sym symbol,\n" +
+                        "    bid int,\n" +
+                        "    ask int\n" +
+                        ")  partition by NONE",
+                0,
+                "group by column does not match key column is select statement "
+        );
+    }
+
+    @Test
+    public void testNoopGroupByWith2Syms() throws Exception {
```

Comment:
This must not work, but it does:
```
"select sym1, sym2, avg(bid) avgBid from x where sym1 in ('AA', 'BB' ) group by sym2"
```

Reply:
just added a test to assertFailure on that query and the test passes